ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -185,111 +175,124 @@
         .withClientSslSupport();
   }
 
-  private static Map<String, Object> commonConfigs(Map<String, Object> props) {
-    return props.entrySet().stream()
-        .filter(e -> !e.getKey().startsWith(KSQL_CONFIG_PROPERTY_PREFIX))
-        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  private static void applyPrefixedConfig(final Map<String, Object> props,
+                                          final String prefix,
+                                          final String key,
+                                          final Object value) {
+    props.put(
+        key.startsWith(prefix) ? key.substring(prefix.length()) : key,
+        value
+    );
   }
 
-  private static Map<String, Object> propertiesWithPrefix(
-      Map<String, Object> props, String prefix) {
-    AbstractConfig abstractConfig = new AbstractConfig(new ConfigDef(), props);
-    return abstractConfig.originalsWithPrefix(prefix);
+  private static boolean streamsConfigFilter(final String config) {
```

Comment:
maybe rename to `isStreamsConfig`.

Reply:
This is going to get changed up a bit in the follow-on PR.